raiden-ts/src/helpers.ts

```diff
@@ -341,3 +344,51 @@ export const isValidUrl = (url: string): boolean => {
       : /^(?:(http|https):\/\/)?([^\s\/$.?#&"']+\.)*[^\s\/$?#&"']+(?:(\d+))*$/;
   return regex.test(url);
 };
+
+/**
+ * Construct entire ContractsInfo using UserDeposit contract address as entrypoint
+ *
+ * @param provider - Ethers provider to use to fetch contracts data
+ * @param userDeposit - UserDeposit contract address as entrypoint
+ * @returns contracts info, with blockNumber as block of first registered tokenNetwork
+ */
+export async function fetchContractsInfo(
+  provider: JsonRpcProvider,
+  userDeposit: Address,
+): Promise<ContractsInfo> {
+  const userDepositContract = UserDepositFactory.connect(userDeposit, provider);
+
+  const oneToN = (await userDepositContract.one_to_n_address()) as Address;
+  const monitoringService = (await userDepositContract.msc_address()) as Address;
+  const monitoringServiceContract = MonitoringServiceFactory.connect(monitoringService, provider);
```

Comment:
I have some issues with the details of naming here. First I thought it is okay to have the implication that all these variables will include the address. At least since the typehint is telling you that it is an address. But then there is the need to differ between the the address and the contract. The solution to only add an appendix to one of them is unsatisfying. Since I appreciate that you split this into two clear readable lines, even if the address retrieval is only necessary for the contract instantiation. But if you feel like this is absolutely horrible to append to all address variables `*Address` and want to stick to the implicit address property, then I would put this address retrieval inline and let the formatter split the code.

Reply:
This pattern is followed all through codebase, because we use the addresses much more often than the contracts (specially for `token` and `tokenNetwork`), which would make it very wasteful to rename everything, and also they `Address` is already implied by the variable type hint, but for the `Contract`, we need to deduplicate the name, and also the name of the contract is e.g. just `MonitoringService` (which inherits `Contract`, from `TypeChain`), which wouldn't tell you it's a contract.
Also, this isn't used only on contract instantiation, but also below, on th returned contractsInfo mapping, so I can't put it in the same line either.